mautic/mautic
# Bug emoji serialization fix

Please answer the following questions. 

| Q  | A
| --- | ---
| Bug fix? | y
| New feature? | n
| Related user documentation PR URL | na
| Related developer documentation PR URL | na
| Issues addressed (#s or URLs) | probably
| BC breaks? | n
| Deprecations? | n

### Required
#### Description:

PR does two things:

1. It changes Doctrine's ArrayType to return an array if serialization is corrupted likely due to a outlier emoji slipping in through a contact's social profile cache. Basically returns an empty array now instead of throwing a 500/exception making Mautic not useable. 

2. It reorganizes the MASSIVE EmojiHelper into individual maps because it was murdering PhpStorm as one massive file. Also changed it to attempt to strip out any missed emojis on HTML or short emoji that could cause serialization issues. 

#### Steps to test this PR:
1. Manually edit a contact's db record and change the social cache to something like `a:1:{}`. Before the PR, you should get an exception or a 500 depending on the environment when viewing the contact list (if it's present on the page) or the contact's details. After applying the PR, you should be able to view the contact without a 500.
2. Create a new email and add some emoji in the subject and body and save. It should display the emoji HTML image. Sending it should result in the actual emoji in the email (unicode).
